Tests/FluentAssertions.Equivalency.Specs/AssertionRuleSpecs.cs

```diff
@@ -245,6 +245,72 @@ public void When_the_runtime_type_does_match_the_equality_comparer_type_it_shoul
         act.Should().Throw<XunitException>().WithMessage("*ConcreteClassEqualityComparer*");
     }
 
+    [Fact]
+    public void When_the_compile_time_type_is_nullable_but_the_equality_comparer_type_is_not_it_should_use_the_default_mechanics()
+    {
+        // Arrange
+        var subject = new { Timestamp = (DateTime?)22.March(2020) };
+
+        var expectation = new { Timestamp = (DateTime?)1.January(2020) };
+
+        // Act
+        Action act = () => subject.Should().BeEquivalentTo(expectation,
+            opt => opt.Using(new DateTimeByYearComparer()));
+
+        // Assert
+        act.Should().Throw<XunitException>().WithMessage("Expected*Timestamp*2020*");
+    }
+
+    [Fact]
+    public void When_the_compile_time_type_is_not_nullable_but_the_equality_comparer_type_is_it_should_use_the_default_mechanics()
+    {
+        // Arrange
+        var subject = new { Timestamp = 22.March(2020) };
+
+        var expectation = new { Timestamp = 1.January(2020) };
+
+        // Act
+        Action act = () => subject.Should().BeEquivalentTo(expectation,
+            opt => opt.Using(new NullableDateTimeByYearComparer()));
+
+        // Assert
+        act.Should().Throw<XunitException>().WithMessage("Expected*Timestamp*2020*");
+    }
+
+    [Fact]
+    public void When_the_runtime_time_type_match_the_equality_comparer_type_it_should_use_the_comparer()
+    {
+        // Arrange
+        var subject = new { Timestamp = (DateTime?)22.March(2020) };
+
+        var expectation = new { Timestamp = (DateTime?)1.January(2020) };
+
+        // Act
+        Action act = () => subject.Should().BeEquivalentTo(expectation, opt => opt
+            .RespectingRuntimeTypes()
+            .Using(new DateTimeByYearComparer()));
+
+        // Assert
+        act.Should().NotThrow();
```

Comment:
We don't explicitly check a not throwing assertion to not throw anymore.
If it throws, it throws anyway..